data_structures/binary_search_tree.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * \file
+ * \brief A simple tree implementation using structured nodes
+ *
+ * \todo update code to use C++ STL library features and OO structure
+ * \warning This program is a poor implementation - C style - and does not
+ * utilize any of the C++ STL features.
+ */
+#include <iostream>
+
+struct node {
+    int val;
+    node *left;
+    node *right;
+};
+
+struct queue {
+    node *t[100];
+    int front;
+    int rear;
+};
+
+queue q;
+
+void enqueue(node *n) { q.t[q.rear++] = n; }
+
+node *dequeue() { return (q.t[q.front++]); }
+
+void Insert(node *n, int x) {
+    if (x < n->val) {
+        if (n->left == NULL) {
+            node *temp = new node;
+            temp->val = x;
+            temp->left = NULL;
+            temp->right = NULL;
+            n->left = temp;
+        } else {
+            Insert(n->left, x);
+        }
+    } else {
+        if (n->right == NULL) {
+            node *temp = new node;
+            temp->val = x;
+            temp->left = NULL;
+            temp->right = NULL;
+            n->left = temp;
+        } else {
+            Insert(n->right, x);
+        }
+    }
+}
+
+int findMaxInLeftST(node *n) {
+    while (n->right != NULL) {
+        n = n->right;
+    }
+    return n->val;
+}
+
+void Remove(node *p, node *n, int x) {
```

Comment:
somewhere we are following camel casing for function names and somewhere not, why is this inconsitency?

Reply:
Again, these files are already existing in the repository. I am trying to fix them. My focus is to make documentation consistent and the code compilable.

Reply:
okay no problem i will just review them as soon as possible we can fix these things later on after merging this big pull request